app/queries/servers/user.ts

```diff
@@ -56,6 +56,10 @@ export const queryUsersById = (database: Database, userIds: string[]) => {
     return database.get<UserModel>(USER).query(Q.where('id', Q.oneOf(userIds)));
 };
 
+export const observeUsersById = (database: Database, userIds: string[]) => {
+    return queryUsersById(database, userIds).observe();
+};
```

Comment:
We prefer not to do this, because usually you want to observe with certain columns (if not, the relevant changes may not trigger), so we leave the caller to decide what to do with the query (in this case, `observeWithColumns`).

Reply:
`observeCurrentUser` gets one user model, and then calls `observe` over that model. Therefore, any change in the model is observed.

What you are doing here is observing a query, therefore... observing a list of elements. Changes in the element won't retrigger the observable.

That being said, this function is expecting to return one single user (even if now it is returning a list), so probably the best solution would be something like:
```suggestion
export const observeUsersById = (database: Database, userIds: string[]) => {
    return queryUsersById(database, userIds).observe().pipe(switchMap((u) => u[0] ? u[0].observe() : of$(undefined)));
};
```

Reply:
> That being said, this function is expecting to return one single user (even if now it is returning a list), 

We do want it to return a list. Given that, totally open to another solution. (But I think it _should_ work as is... See: https://github.com/mattermost/mattermost-mobile/blob/3abaf8893d359694edd425a44e0f48ff224c75d9/app/components/post_list/combined_user_activity/index.ts#L36)

Reply:
what larkox said

Reply:
Ok, makes sense, thanks! Tested and works when changing name and avatar, nice. Also, with one more check in `Channel` we can be sure `JoinCallBanner` is only mounted when needed. Done here: 732e57e9dfadaebfc501f468ff5e350732b79610